pdf_viewer broken in 4.0.189 when using Webpack

Attach (recommended) or Link to PDF file here: No PDF needed; the viewer does not load properly.

Configuration:
- Web browser and its version: Chrome 119.0.6045.105
- Operating system and its version: Linux (Nobara 38)
- PDF.js version: 4.0.189
- Is a browser extension: No

It seems that https://github.com/mozilla/pdf.js/commit/927e50f5d48a67e76f2a51c112ea5a98867822fe broke `pdf_viewer`'s usage of `pdfjsLib` when using Webpack. Previously, it was possible to use it like this:
```js
import * as pdfjs from "pdfjs-dist/legacy/build/pdf";
import { EventBus, PDFLinkService, PDFSinglePageViewer } from "pdfjs-dist/legacy/web/pdf_viewer";
```
However, this now breaks as `pdf.mjs` now contains `__webpack_exports__ = globalThis.pdfjsLib = await __webpack_exports__;`. This means that when the `pdf_viewer` code executes, `pdfjsLib` is not set yet.

Please let me know if I'm missing anything, but I believe this is a regression in the library, as I wasn't able to find a way to use `pdf_viewer` with Webpack now.

Just trying out v4 in a new project and getting:

```
[watch] build started (change: "js/pdf_preview/index.js")
✘ [ERROR] Top-level await is currently not supported with the "iife" output format

    node_modules/pdfjs-dist/build/pdf.mjs:16837:53:
      16837 │ ...ck_exports__ = globalThis.pdfjsLib = await __webpack_exports__;
            ╵                                         ~~~~~
```

See https://elixirforum.com/t/importing-pdf-mjs-into-app-js/59578

--- a/gulpfile.mjs
+++ b/gulpfile.mjs
@@ -422,7 +422,7 @@ function tweakWebpackOutput(jsName) {
       case " __webpack_exports__ = {};":
         return ` __webpack_exports__ = globalThis.${jsName} = {};`;
       case " __webpack_exports__ = await __webpack_exports__;":
-        return ` __webpack_exports__ = globalThis.${jsName} = await __webpack_exports__;`;
+        return ` __webpack_exports__ = globalThis.${jsName} = await (globalThis.${jsName}Promise = __webpack_exports__);`;
     }
     return match;
   });

--- a/web/pdfjs.js
+++ b/web/pdfjs.js
@@ -13,6 +13,15 @@
  * limitations under the License.
  */
 
+// Ensure that the viewer waits for the library to complete loading,
+// to avoid breaking e.g. the standalone viewer components (see issue 17228).
+if (
+  (typeof PDFJSDev === "undefined" || PDFJSDev.test("GENERIC")) &&
+  !globalThis.pdfjsLib
+) {
+  await globalThis.pdfjsLibPromise;
+}
+
 const {
   AbortException,
   AnnotationEditorLayer,

